internal/elasticsearch/index/commons.go

```diff
@@ -80,3 +80,30 @@ func FlattenIndexAlias(name string, alias models.IndexAlias) (interface{}, diag.
 
 	return a, diags
 }
+
+func ExpandLifecycle(definedLifecycle *schema.Set) (*models.LifecycleSettings, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	lifecycleMap := definedLifecycle.List()[0].(map[string]interface{})
+
+	if lifecycleMap != nil {
+		lifecycle := &models.LifecycleSettings{}
+		if s, ok := lifecycleMap["data_retention"]; ok {
+			lifecycle.DataRetention = s.(string)
+		}
+		return lifecycle, diags
+	}
+	return nil, diags
+}
+
+func FlattenLifecycle(lifecycle *models.LifecycleSettings) (interface{}, diag.Diagnostics) {
```

Comment:
```suggestion
func FlattenLifecycle(lifecycle *models.LifecycleSettings) interface{} {
```

Similarly, diags is unused.